File: src/main/java/org/example/collection/validators/ValidatorLocationName.java
package org.example.collection.validators;

import org.example.exception.InvalidDataError;

/**
 * Валидатор названия локации
 */
public class ValidatorLocationName implements Validator<String> {
    /**
     * Проверяет строку на пустоту
     *
     * @param s строка
     * @throws InvalidDataError если строка пуста
     */
    @Override
    public void valide(String s) throws InvalidDataError {
        if (s == null) {
            return;
        }
        if (s.isBlank() | s.isEmpty()) {
            throw new InvalidDataError();
        }
    }
}

File: common/src/main/kotlin/collection/validators/ValidatorLocationName.kt
package collection.validators

import exeption.InvalidDataError


class ValidatorLocationName : IValidator<String?> {

    override fun valid(t: String?) {
        if (t == null) {
            return
        }
        if (t.isBlank() or t.isEmpty()) {
            throw InvalidDataError()
        }
    }
}